extensions/positron-catalog-explorer/src/catalogs/snowflake.ts

```diff
@@ -29,59 +73,179 @@ export function registerSnowflakeProvider(
 }
 
 /**
- * Register a Snowflake catalog provider using the well-known authentication
- * provider.
+ * Register a Snowflake catalog provider using SSO authentication.
  */
-async function registerSnowflakeCatalog(
-	_context: vscode.ExtensionContext,
+export async function registerSnowflakeCatalog(
+	context: vscode.ExtensionContext,
+	account?: string,
 ): Promise<CatalogProvider | undefined> {
 	try {
-		// Unfortunately, authentication.getSession() currently hangs
-		// for ~5s when no Snowflake authentication provider is
-		// registered. In order to make this less confusing for users,
-		// show a "progress" notification when this hang occurs.
-		const session = await vscode.window.withProgress(
+		// Get recent account names to suggest to the user
+		const recentAccounts = context.globalState.get<string[]>(RECENT_SNOWFLAKE_ACCOUNTS_KEY) || [];
+
+		// User is not re-authenticating an existing account, prompt for account name
+		if (!account) {
+			// If we have recent accounts, show a quick pick first
+			if (recentAccounts.length > 0) {
+				// Add a "New Account" option at the end
+				const items = [
+					...recentAccounts.map(name => ({
+						label: name,
+						description: 'Recent account'
+					})),
+					{
+						label: 'Enter New Account...',
+						description: 'Provide a different account identifier'
+					}
+				];
+
+				const selection = await vscode.window.showQuickPick(items, {
+					placeHolder: 'Select a recent account or enter a new one',
+					ignoreFocusOut: true
+				});
+
+				if (!selection) {
+					return undefined; // User canceled
+				}
+
+				if (selection.label === 'Enter New Account...') {
+					// User wants to enter a new account
+					account = await vscode.window.showInputBox({
+						prompt: 'Enter your Snowflake account identifier',
+						placeHolder: 'orgname-accountname',
+						validateInput: (value) => {
+							return value.trim() === '' ? 'Account name cannot be empty' : null;
+						},
+						ignoreFocusOut: true
+					});
+				} else {
+					// User selected an existing account
+					account = selection.label;
+				}
+			} else {
+				// No recent accounts, just show the input box
+				account = await vscode.window.showInputBox({
+					prompt: 'Enter your Snowflake account identifier',
+					placeHolder: 'orgname-accountname',
+					validateInput: (value) => {
+						return value.trim() === '' ? 'Account name cannot be empty' : null;
+					},
+					ignoreFocusOut: true
+				});
+			}
+		}
+
+		// If still no account, user canceled
+		if (!account) {
+			return undefined;
+		}
+
+		// Place snowflake logs in users workspace folder if available
+		const config = vscode.workspace.getConfiguration('catalogExplorer');
+		const logLevelStr = config.get<string>('logLevel', 'INFO') as SnowflakeLogLevel;
+
+		snowflake.configure({
+			logLevel: logLevelStr,
+			logFilePath: vscode.workspace.workspaceFolders
```

Comment:
As follow up work, we will want to send these logs to the output channel. We'll need to set up a file watcher + streamer similar to [kallichore's LogStreamer](https://github.com/posit-dev/positron/blob/main/extensions/positron-supervisor/src/LogStreamer.ts#L0-L1) to stream everything. For now, we place the file in a user's workspace if there is one available.